packages/realm-network-transport/src/NetworkTransport.ts

```diff
@@ -1,7 +1,7 @@
 declare var process: any;
 declare var require: ((id: string) => any) | undefined;
```

Comment:
The indent of the "realm-network-transport" package is to provide a cross-platform / universal package thats independent of the runtime.

If we es-imported "node-fetch" here we get in trouble when requiring "realm-network-transport" from a project that doesn't provide the node.js runtime types (like Webpack for a web-project). Either that or all the global node.js types would get pulled into the users project ([as node-fetch's types tribleslash-references the node types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node-fetch/index.d.ts#L17)).

We cannot require "node-fetch" in the beginning of the file, because that wouldn't make sense when using the package to transpile to be used in a web-browser or React Native - which is also why we're hiding the require from static analysis (webpack, ReactNativess haste map, etc).

Reply:
I mean why not do 
```
const node_fetch;
if (isNodeProcess) {
    node_fetch= node_require();
}
```
in the beginning. 

got you on the hiding of the import.

Reply:
I don't think I'm following you entirely here, I want to check the type of `require` to help the TypeScript type checker understand that this is a callable function (and not undefined, as the type declaration also allows - in the case of a browser).

Reply:
What I am saying is that any require should be at the top of the file https://github.com/realm/realm-js/blob/796df0c0098a556abfac9fb019e0e12a83183d68/packages/realm-network-transport/src/NetworkTransport.ts#L75
In this case I see that there is a check for that in the beginning of the constructor
https://github.com/realm/realm-js/blob/796df0c0098a556abfac9fb019e0e12a83183d68/packages/realm-network-transport/src/NetworkTransport.ts#L68 so it's not an issue. Either way if we can move these require calls to the top its better IMO. 
Let's leave it as it is in this case.